File: 1302-delete-characters-to-make-fancy-string/1302-delete-characters-to-make-fancy-string.cpp
class Solution {
public:
    string makeFancyStringRemove(string s) {
        char previous = '\0';
        bool twice = false;
        string ret = "";
        for(string::iterator it = s.begin(); it != s.end();) {
            char c = *it;
            if (c == previous) {
                if (twice) {
                    s.erase(it);
                } else {
                    twice = true;
                    it++;
                }
            } else {
                previous = c;
                twice = false;
                it++;
            }
        }
        return s;
    }

    string makeFancyString(string s) {
        char previous = '\0';
        bool twice = false;
        string ret = "";
        for(string::iterator it = s.begin(); it != s.end();it++) {
            char c = *it;
            if (c == previous) {
                if (!twice) {
                    ret.push_back(c);
                    twice = true;
                }
            } else {
                ret.push_back(c);
                previous = c;
                twice = false;
            }
        }
        return ret;
    }
};

File: 1302-delete-characters-to-make-fancy-string/1302-delete-characters-to-make-fancy-string.go
func makeFancyString(s string) string {
	if len(s) == 0 {
		return s
	}

	var ret []rune
	var previous rune
	count := 0

	for _, c := range s {
		if c == previous {
			if count < 2 {
				ret = append(ret, c)
				count++
			}
		} else {
			ret = append(ret, c)
			previous = c
			count = 1
		}
	}

	return string(ret)
}